packages/client-utils/src/sync-tools.js

```diff
@@ -305,21 +305,16 @@ export const waitUntilInvitationReceived = (addr, io, options) => {
 
 /**
  * @param {string} basePath
- * @param {{follow: (...params: string[]) => Promise<object>; marshaller: import("@endo/marshal").Marshal<any>}} io
+ * @param {import('./vstorage-kit').VstorageKit} vstorage
  * @returns {Promise<ElectionResult>}
  */
-export const fetchLatestEcQuestion = async (basePath, io) => {
-  const { follow, marshaller } = io;
-  const pathOutcome = `:${basePath}.latestOutcome`;
-  const pathQuestion = `:${basePath}.latestQuestion`;
+export const fetchLatestEcQuestion = async (basePath, vstorage) => {
+  const pathOutcome = `${basePath}.latestOutcome`;
+  const pathQuestion = `${basePath}.latestQuestion`;
 
   const [latestOutcome, latestQuestion] = await Promise.all([
-    follow('-lF', pathOutcome, '-o', 'text').then(outcomeRaw =>
-      marshaller.fromCapData(JSON.parse(outcomeRaw)),
-    ),
-    follow('-lF', pathQuestion, '-o', 'text').then(questionRaw =>
-      marshaller.fromCapData(JSON.parse(questionRaw)),
-    ),
+    /** @type {Promise<ElectionResult["latestOutcome"]>} */ (vstorage.readLatestHead(pathOutcome)),
```

Comment:
`readPublished` produces the type without casting, but that requires `basePath` to be a string template.

I consider this tech debt but accept it as expedient for the scope of this PR. However, please don't export `fetchLatestEcQuestion` so that more stuff isn't built on this debt

Reply:
> readPublished produces the type without casting, but that requires basePath to be a string template.

Oh I wasn't aware of; https://github.com/Agoric/agoric-sdk/blob/ff9b882dd93e3f8d3b7e9d305bbafda6bfb17fac/packages/client-utils/src/types.d.ts#L25

Thanks.

> However, please don't export fetchLatestEcQuestion so that more stuff isn't built on this debt

Understood. On it.

Reply:
Resolved in https://github.com/Agoric/agoric-sdk/pull/10385/commits/e1eaa5bf86ea936dee10bd996a60a49e5d157694